contracts/peripheral/PoolProxy.sol

```diff
@@ -192,11 +40,22 @@ contract PoolProxy is DecimalMath {
     /// @param maxFYDai maximum amount of fyDai to be borrowed to mint liquidity. 
     /// @return The amount of liquidity tokens minted.  
     function addLiquidity(IPool pool, uint256 daiUsed, uint256 maxFYDai) public returns (uint256) {
-        onlyKnownPool(pool);
         IFYDai fyDai = pool.fyDai();
         require(fyDai.isMature() != true, "YieldProxy: Only before maturity");
         require(dai.transferFrom(msg.sender, address(this), daiUsed), "YieldProxy: Transfer Failed");
 
+        // Allow the Treasury to take chai when posting
```

Comment:
Help me understand these lines. Previously, we were just giving blanket approvals from the Proxy to various contracts in the constructor, right? Now, it appears that you are giving these approvals if users haven't already given the approval? Why the need for the check?

Reply:
The YieldProxy family of contracts needs sometimes to hold some tokens, before a Pool or the Treasury takes them using `transferFrom`. Before we could set these approvals in the YieldProxy constructor, which would result in something like the Dai contract recording `allowance[YieldProxy.address][Treasury.address] = MAX`.

Now the contract executing the functions in YieldProxy is not YieldProxy itself. The DSProxy contract for user X will be executing `repayWithDai` or whatever, meaning that the Dai contract needs to have `allowance[DSProxyForUserX.address][Treasury.address] = MAX`

That means that those approvals need to be set up for each DSProxy. Since we can't change DSProxyFactory or DSProxyRegistry, we can't allow anyone except the user to set those approvals, and we don't want a separate `init` transaction, we check if the approvals are set with each call, and set them if they are not.

Reply:
Why not just always set them? Is this more gas-efficient?

Reply:
Yes, just checking the value is 1500 gas, approving would be around 15K.